scripts/dts/python-devicetree/src/devicetree/grutils.py

```diff
@@ -132,6 +132,40 @@ def scc_order(self):
         return self.__scc_order
     __scc_order = None
 
+    @staticmethod
+    def _estimate_fanout(edge_map, root):
+        # Compute an estimate of the maximum fan-out of the edge_map from a given root.
```

Comment:
Can you give a writeup of the algorithm a bit more abstractly along with more of a formal "algorithms 101" argument why it is computing an upper bound on the correct fanout? I'd like to see that recorded in the commit log for when we get back to this later.

Reply:
Alright, I've had a go at improving the explanation.
I never took "Algorithms {N}", so let me know if it's missing some essential component of an algorithm description :)

Reply:
Thanks! The observation that the number we want to compute is the number of leaf nodes if the graph is a tree makes sense. but I'm not totally sure about what you mean for observations 2. and 3. I'm still thinking this over as a result. (If you have any friends who are good at graph problems who might be able to provide a proof that it's intractable or a clever algorithm for finding the answer, please ask them also...)

Reply:
Conceptually its a transform from a DAG to a N-ary tree. 2 and 3 are trying to explain that the transform operation cannot reduce the actual maximum fanout. Once you get to an N-ary tree, the maximum fanout is easy to compute, but due to the transforms, it can be somewhat higher than the fanout for the original graph.